spark/src/test/scala/org/apache/spark/sql/delta/DeltaTableCreationTests.scala

```diff
@@ -2599,6 +2599,19 @@ class DeltaTableCreationSuite
       }
     }
   }
+
+
+  test("Creating a table with catalogOwned feature set should be blocked") {
+    val testTableName = "test_table"
+    withTable(testTableName) {
+      val e = intercept[DeltaUnsupportedOperationException] {
+        sql(
+          s"""CREATE TABLE $testTableName (id INT)
+             |USING DELTA
+             |TBLPROPERTIES('delta.feature.catalogOwned-preview' = 'supported')""".stripMargin)
```

Comment:
In case the table feature name changes in the future.
```suggestion
             |TBLPROPERTIES('delta.feature.${CatalogOwnedTableFeature.name}' = 'supported')""".stripMargin)
```

Reply:
I removed the test for now as I am allowing UTs to bypass the check. I'll add an e2e test for it.